include/AST/Types/FloatTy.hpp

```diff
@@ -0,0 +1,29 @@
+#ifndef GLU_AST_TYPES_FLOATTY_HPP
+#define GLU_AST_TYPES_FLOATTY_HPP
+
+#include "TypeBase.hpp"
+
+namespace glu::types {
+class FloatTy : public TypeBase {
+    unsigned _bitWidth;
+
+public:
+    /// @brief Constructor for the FloatTy class.
+    FloatTy(unsigned bitWidth) : TypeBase(TypeKind::FloatTyKind), _bitWidth(bitWidth) {
+        assert(bitWidth > 0 && "Bit width must be greater than 0");
+    }
+
+    unsigned getBitWidth() const { return _bitWidth; }
```

Comment:
maybe isSingle (isFloat, 32 bits), isHalf (16), isDouble (64), isIntelLongDouble (80)?
and static constants for those numbers to make constructing easier, for example `FloatTy(FloatTy::DOUBLE)`